metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java

```diff
@@ -238,8 +239,18 @@ private UpdateAspectResult wrappedIngestAspectToLocalDB(@Nonnull final Urn urn,
       @Nonnull List<Pair<String, RecordTemplate>> aspectRecordsToIngest,
       @Nonnull final AuditStamp auditStamp, @Nonnull final SystemMetadata providedSystemMetadata) {
     validateUrn(urn);
+    aspectRecordsToIngest.forEach(pair -> validateAspect(urn, pair.getSecond()));
     return ingestAspectsToLocalDB(urn, aspectRecordsToIngest, auditStamp, providedSystemMetadata);
   }
+
+  private void validateAspect(Urn urn, RecordTemplate aspect) {
+    UrnValidator validator = new UrnValidator(_entityRegistry);
```

Comment:
Instead we should call this one 

EntityUrnValidator or something similar

Reply:
Sure, wasn't sure on the naming, I renamed it a few times before pushing. The other one does a lighter check, but requires less dependencies. It might could be removed, but the usages are embedded in a few flows where we don't have full access to the entity registry

Reply:
Renamed to EntityRegistryUrnValidator